### ISSUE TITLE
No unauthenticated calls to the web api

### ISSUE THREAD

**Issue by Funkbone** [JohnnyCrazy/SpotifyAPI-NET#149]
Just discovered my app was returning null for every search and discovered this

https://developer.spotify.com/news-stories/2017/01/27/removing-unauthenticated-calls-to-the-web-api/

Basically nothing works so your getting started guide is no longer valid:

        _spotify = new SpotifyWebAPI()
        {
            UseAuth = false, //This will disable Authentication.
        }
        FullTrack track = _spotify.GetTrack("3Hvu1pq89D4R0lyPBoujSv");
        Console.WriteLine(track.Name);

I've looked at doing the authorization stuff but so far it's over my head.  Any quick and dirty way around it?

Every call gets this

{
  "error" : {
    "status" : 401,
    "message" : "No token provided"
  }
}

**Comment by JohnnyCrazy** [comment]
Fastest way is https://johnnycrazy.github.io/SpotifyAPI-NET/SpotifyWebAPI/auth/#clientcredentialsauth (`UseAuth` has to be true tho, need to update the docs at some point)

There currently is no safe way of using this library on client-side. Even ImplicitGrantAuth is kinda unsafe because you have to whitelist `localhost` as a redirect URI.

**Comment by Funkbone** [comment]
I found this guide here https://hendrikbulens.wordpress.com/2015/01/07/c-and-the-spotify-web-api-part-i/ which uses an encoded string for the clientid/secret and it looks like it's working to get the token, at least in fiddler.  Is that "safe"?

Well it's not working anyway I'm getting a new error:
 { "error": { "status": 400, "message": "Only valid bearer authentication supported" } }

I'll try your way and get back to you

**Comment by Funkbone** [comment]
OK I got it with your ClientCredentialsAuth().  Besides the UseAuth = true I also had to change the Scope line you have in the example to Scope = Scope.UserReadPrivate; instead of Scope.USER_READ_PRIVATE,

I know it's outside the scope of this issue or even your api but it would also be helpful if there was a guide to set up the localhost redirect uri in a C# application.  I'm not getting it but I don't really know what I'm doing either.  I'm self taught and still have to google everything I do.

Anyway thanks for the help! I guess this issue can be closed

**Comment by JimmyAppelt** [comment]
@Funkbone the information about localhost redirect you seek, can be found in the documentation here: https://johnnycrazy.github.io/SpotifyAPI-NET/SpotifyWebAPI/auth/

The change of `Scope.UserReadPrivate;` is indeed correct.

**Comment by Funkbone** [comment]
I was getting ready to complain about the documentation but decided to take another crack at it and wow I finally figured out the damn ImplicitGrantAuth.  It really isn't as complicated as I was trying to make it.  I tried this a long time ago and apparently was setting UseAuth to false in another forgotten place so the token wasn't working when I called it and I was banging my head against the wall for weeks!  Finally!  Sorry for my confusion and thanks again for all your work on this and your help.
